compiler/noirc_frontend/src/elaborator/path_resolution.rs

```diff
@@ -110,12 +110,36 @@ enum MethodLookupResult {
     FoundMultipleTraitMethods(Vec<(TraitId, Ident)>),
 }
 
+/// Determines whether datatypes found along a path are to be marked as referenced
```

Comment:
Please note the cases where a type is constructed v.s. referenced in the docs

Reply:
Makes sense. Done!